.github/workflows/build-docker.yml

```diff
@@ -0,0 +1,68 @@
+name: Build Docker image
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+env:
+  ARM_GNU_TOOLCHAIN_VERSION: 14.2.rel1
+  ARM_GNU_TOOLCHAIN_PLATFORM: x86_64-arm-none-eabi
+  EM_VERSION: 4.0.1
+  EM_NODE_VERSION: 20.18.0_64bit
+
+jobs:
+  build-docker:
+    runs-on: ubuntu-24.04
+    permissions:
+      packages: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      
+      - name: Cache submodules
+        id: cache-submodules
+        uses: actions/cache@v4
+        with:
+          path: |
+            third_party/cmsis_core/CMSIS
+            third_party/hal_stm32/stm32-sdk
+            third_party/nimble/mynewt-nimble
+            third_party/ti_bt_sp/ti-service-packs
+            third_party/hal_nordic/nrfx
+            third_party/freertos/FreeRTOS-Kernel
+            third_party/nanopb/nanopb
+            third_party/tinymt/TinyMT
+          key: ${{ runner.os }}-submodules
+
+      - name: Init submodules
+        if: steps.cache-submodules.outputs.cache-hit != 'true'
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
```

Comment:
question: why is this needed?

Reply:
To take advantage of `cache-from` and `cache-to` in Docker build steps